### PR TITLE
Sets RLL to configure reconnection with LL

### LINKED ISSUE #82: Add resuming support
Implementation details can be found at: https://github.com/Frederikam/Lavalink/blob/master/IMPLEMENTATION.md#resuming-lavalink-sessions

A similar implementation is proposed in a PR to an alternate library we don't use yet, Wavelink.

This would need to be added to RLL to resolve the issues users have reported where a bot will be connected, playing, and then stop sending voice data. Possibly an event like WEBSOCKET_CLOSED would also be needed to be added to the lib as well, and/or handling on the node to know wether to close all players on the node if the session isn't resumed, but I'll leave those choices up to anyone that is interested in attempting to implement this.

### PR DESCRIPTION
**Pull request by Drapersniper**
Enabled Track resuming when the WS is disconnected, which should in theory stop us seeing the track is stuck error.

closes #82

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,10 @@ cache: pip
 notifications:
```
Comment (Jackenmen) [inline .travis.yml]: We've moved to GitHub Actions and don't use Travis anymore, this is here just for archive purpose I guess. Please revert the changes to this file.

File: lavalink/enums.py [hunk 2]
```diff
@@ -35,6 +35,9 @@ class LavalinkEvents(enum.Enum):
     TRACK_START = "TrackStartEvent"
     """The track playback started."""
 
+    WebSocketClosedEvent = "WebSocketClosedEvent"
```
Comment (Jackenmen) [inline lavalink/enums.py]: (Considering that this library is coming to an end either way, this is a nitpick and can be ignored)
Why not call this `WEBSOCKET_CLOSED` similarly to other events?

File: lavalink/node.py [hunk 3]
```diff
@@ -117,6 +162,33 @@ def __init__(
...
             _nodes.append(self)
 
+        self._closers = (
+            aiohttp.WSMsgType.CLOSE,
+            aiohttp.WSMsgType.CLOSING,
+            aiohttp.WSMsgType.CLOSED,
+        )
+
+    @property
+    def headers(self) -> dict:
```
Comment (Jackenmen) [inline lavalink/node.py:171]: Why not just use `return self._get_connect_headers()` in this property?

File: lavalink/node.py [hunk 4]
```diff
@@ -87,8 +124,10 @@ def __init__(
             User ID of the bot.
         num_shards : int
             Number of shards to which the bot is currently connected.
-        ready : asyncio.Event
-            Set when the connection is up and running, unset when not.
+        resume_key : Optional[str]
+            A resume key used for resuming a session upon re-establishing a WebSocket connection to Lavalink.
+        resume_timeout : inr
```
Comment (Jackenmen) [inline lavalink/node.py]:
```suggestion
        resume_timeout : int
```

File: lavalink/node.py [hunk 5]
```diff
@@ -199,12 +293,9 @@ async def listener(self):
...
                 if msg.type is aiohttp.WSMsgType.ERROR:
                     log.debug("Ignoring exception in listener task", exc_info=msg.data)
-                elif msg.type in (
-                    aiohttp.WSMsgType.CLOSED,
-                    aiohttp.WSMsgType.CLOSING,
-                    aiohttp.WSMsgType.CLOSE,
-                ):
+                elif msg.type in self._closers:
                     log.debug("Listener closing: %s", msg.extra)
+                    await self._reconnect()
```
Comment (Jackenmen) [inline lavalink/node.py]: There's a break after this `if` which results in `_reconnect()` getting awaited here first and later after leaving the `while` loop, a new task for `_reconnect()` gets spawned. If I'm getting this right, this means we end up reconnecting twice and possibly also end up with duplicated websocket connection and two listener() tasks as `connect()` does not have anything preventing it from connecting when it's already connected.

**Review by Jackenmen — DISMISSED** [review]
Code review.

A functional review will be handled by aikaterna.

**Review by Jackenmen — COMMENTED** [review]
Looks good to me on the code review part (aside from that weird `_Key` class that could be simplified, but it shouldn't break anything so it's fine).

For this to be approved, we will also need a functional review that's gonna be made later by aikaterna.

File: lavalink/node.py [hunk 6]
```diff
@@ -135,36 +199,55 @@ async def connect(self, timeout=None):
...
-    def _get_connect_headers(password, user_id, num_shards):
-        return {"Authorization": password, "User-Id": str(user_id), "Num-Shards": str(num_shards)}
+    def _get_connect_headers(self) -> dict:
+        headers = {
+            "Authorization": self.password,
+            "User-Id": str(self.user_id),
+            "Num-Shards": str(self.num_shards),
+        }
+        if self._resume_key:
+            headers["Resume-Key"] = self._resume_key
```
Comment (Drapersniper) [inline lavalink/node.py]:
```suggestion
            headers["Resume-Key"] = str(self._resume_key)
```

**Review by aikaterna — APPROVED** [review]

**Review by Jackenmen — APPROVED** [review]
